### PR TITLE
[Feature][datasource] Add datasource api

### PR DESCRIPTION
**Pull request by xiaoqigh**

### CODE REVIEW

File: seatunnel-datasource/seatunnel-datasource-plugins/datasource-plugins-api/src/main/java/org/apache/seatunnel/datasource/plugin/api/DataSourcePluginInfo.java [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* (C)2023 */
```
Comment (EricJoy2048) [inline seatunnel-datasource/seatunnel-datasource-plugins/datasource-plugins-api/src/main/java/org/apache/seatunnel/datasource/plugin/api/DataSourcePluginInfo.java:1]: delete this.

File: seatunnel-server/seatunnel-app/pom.xml [hunk 2]
```diff
@@ -238,5 +238,42 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.5</version>
```
Comment (EricJoy2048) [inline seatunnel-server/seatunnel-app/pom.xml:254]: remove this

File: seatunnel-server/seatunnel-app/pom.xml [hunk 3]
```diff
@@ -238,5 +238,42 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-datasource-client</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.auto.service</groupId>
+            <artifactId>auto-service-annotations</artifactId>
+            <version>1.0.1</version>
```
Comment (EricJoy2048) [inline seatunnel-server/seatunnel-app/pom.xml:265]: same as above.

File: seatunnel-server/seatunnel-app/pom.xml [hunk 4]
```diff
@@ -238,5 +238,42 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>datasource-s3</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
```
Comment (EricJoy2048) [inline seatunnel-server/seatunnel-app/pom.xml:271]: ${project.version}

File: seatunnel-server/seatunnel-app/pom.xml [hunk 5]
```diff
@@ -238,5 +238,42 @@
...
+            <scope>compile</scope>
```
Comment (EricJoy2048) [inline seatunnel-server/seatunnel-app/pom.xml:272]: provided

File: seatunnel-datasource/seatunnel-datasource-plugins/datasource-plugins-api/src/main/java/org/apache/seatunnel/datasource/plugin/api/DataSourcePluginInfo.java (hunk 1 above)
Comment (xiaoqigh) [reply seatunnel-datasource/seatunnel-datasource-plugins/datasource-plugins-api/src/main/java/org/apache/seatunnel/datasource/plugin/api/DataSourcePluginInfo.java:1]: ok.

**Pull request merged**